app/src/main/java/com/example/sharingang/items/ItemsViewModel.kt

```diff
@@ -127,6 +135,29 @@ class ItemsViewModel @Inject constructor(
         }
     }
 
+    /**
+     * Order items in the database
+     *
+     * @param orderBy ORDERING
+     * @param isAscending either ascending or descending order
+     */
+    fun orderItems(orderBy: ORDERING, isAscending: Boolean) {
+        viewModelScope.launch(Dispatchers.IO) {
+            var results = itemRepository.getAll().sortedWith(compareBy {
+                when (orderBy) {
+                    ORDERING.DATE -> it.createdAt
+                    ORDERING.PRICE -> it.price
+                    ORDERING.NAME -> it.title
+                    ORDERING.CATEGORY -> it.category
+                }
+            })
+            if (!isAscending) {
+                results = results.asReversed()
+            }
+            _orderedItems.postValue(results)
```

Comment:
Is this called from a coroutine? `postValue` isn't necessary if not

Reply:
I copied the structure from searchItems

Reply:
Yeah, this is never called from inside a coroutine, so you can just do `_orderedItems.value = results`

Reply:
And at this point we're in `viewModelScope.launch(Dispatchers.IO) { ... }` which would make it inside a coroutine

Reply:
Oh, right